lib/client.js

```diff
@@ -1381,10 +1389,15 @@ function write (client, request) {
     socket[kReset] = true
   }
 
+  if (reset) {
+    socket[kReset] = true
+  }
+
   if (blocking) {
     socket[kBlocking] = true
   }
 
+  const shouldReset = request.reset != null ? request.reset : client[kClientResetValue]
```

Comment:
Nit: `request.reset ?? client[kClientResetValue]`

Reply:
Note: this would drop support for v12 and you'd need to remove it from the ci.

Reply:
Ah, that's true. I recommend doing it in a new pull request, if we are going to drop it (and I think we should)